### ISSUE TITLE
Multiple branches with the same name appearing on their own.

### ISSUE THREAD

**Issue by charliefabley** [FlutterFlow/flutterflow-issues#2866]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Both new and old branches have apparently replicated themselves with the same name. Some of them have different edit times, which is a bit disconcerting

### Expected Behavior

That there was only one branch with one unique name, and that they should not replicate themselves.

### Steps to Reproduce

Go to our project and look at the branches

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4KkvLlw4x2ruZE7rruK/lVtCYWNjsja7sslsJtbwAdNYvpG7l/S8+kfhNuXbC0dAhleWKKgmUK+vPemvHtEvE6CzqWRZxd1s1pEw/gWn+hbq67EajpYG5/De1YN3Ky1qSskSNAJsxhLnhg1ziHNd6RTgnDNr34DDEzAOaaOZOK2SrDX1iXc2URm05KZDPz

### Context

We have we have had numerous problems with marge, that has left us losing hours if not sometimes days of work. This appears just to be another bug in the merged system.

### Visual documentation

![branches](https://github.com/FlutterFlow/flutterflow-issues/assets/153652817/fecdd7ea-7a23-48de-876e-181ad235ba75)

### Additional Info

No additional information

### Environment

```markdown
- FlutterFlow version: 
- Platform: Windows current and web version
- Browser name and version: web version and chrome
- Operating system and version affected: windows 11 and 12
```

### General

- [X] I was collaborating with other users in the same project prior to data loss
- [ ] I navigated between different project views leading up to data loss
- [ ] The UI was running very slowly prior to data loss
- [ ] A crash or other bug occurred prior to data loss
- [ ] Data was not saved in a snapshot
- [ ] I was using flutterflow from multiple devices
- [ ] I was using flutterflow from multiple platforms (e.g. both web and MacOS Desktop app)

### Relative to the time the changes were made, data was lost within

- [ ] 1 minute
- [X] 1 hour
- [ ] 1 day
- [ ] 1 week

### When following my steps to reproduce, data loss happens

- [ ] ~100% of the time
- [X] ~50% of the time
- [ ] ~10% of the time
- [ ] ~1% of the time
- [ ] Very rarely or only encountered once

**Comment by charliefabley** [comment]
I'm not sure that we necessarily did anything, it just happened. It was a
colleague of mine and she reported it to me fairly swiftly, and said that
she had simply created a branch, and came back to me I believe within an
hour of two of it happening. So all I can say is that we did nothing out of
the ordinary. They remained there until today, when I suggested that she
merged the one she'd been working on most recently, and deleted the others.
The merge worked.
